foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/expressions/ArgumentListFunctionExpression.java

```diff
@@ -97,11 +97,10 @@ public void setOperator(ExpressionOperator theOperator) {
      */
     @Override
     public void printSQL(ExpressionSQLPrinter printer) {
-        ListExpressionOperator realOperator;
-        realOperator = (ListExpressionOperator)getPlatformOperator(printer.getPlatform());
```

Comment:
Sorry but this removal is problematic as there should be some differences between DB platforms.
Try e.g. `mvn clean verify -pl :org.eclipse.persistence.jpa.jse.test -P derby` against before modification and after to see what happens. `-P derby` means that Implicit Apache Derby DB is used (it is managed started/stopped by EclipseLink Maven build scripts).

Reply:
Ok, I see what you mean.
In that case the problem is that getPlatformOperator() returns the same cached instance of the operator on each call which leads to errors with the operators with _variable number_ of arguments like "coalesce" and "case when".
I think I have a solution in mind. I will push it later.